### ISSUE TITLE
Strange talk-related problem crashed moon

### ISSUE THREAD

**Issue by whereswaldon** [urbit/urbit#798]
The chain of events leading to this problem is long, and I have no idea at what point the problem was introduced. I'm sorry if this isn't a useful bug report, if it's even a bug.

I was running a moon on my desktop. It automagically synced talk with my planet's talk, which amazed me. Everything was running fine until I tried to activate a message in :talk. You see, I miscounted and activated a message number that didn't have a corresponding message (at least, I think so). This caused talk to crash. I was scared, but I asked @Fang- for help and he suggested variants of:

```
|unlink %talk
|link %talk
|unlink ~planet-name %talk
|link ~planet-name %talk
```
Ultimately, using the planet name did re-establish my ability to use :talk from the moon. The reason I mention it is that I don't know whether I might have run `|unlink %talk` without a corresponding `|link %talk` while trying to solve that. If so, it could be related to what happened later.

Later on, :talk just kinda stopped doing anything on the moon. It would let me type, but acted as though pressing the "Enter" key was invalid. I couldn't do anything. I switched to the dojo (don't remember why), and I started getting this error:
```
[! gall: %hood: diff: no /talk-report/drum/phat/~hasmer-dovbex/talk  [%ap-lame %hood %diff]]
[~lonwet-siglyr-hasmer-dovbex, driving ~lonwet-siglyr-hasmer-dovbex]
```
It repeated ad-infinitum, and so I tried a lot of variants of the above commands with little success. When I tried to C-x back into :talk, there was just empty space where the :talk prompt should have been. The status messages were coming think and fast, so I decided to try to suspend the process with C-z. This apparently killed the urbit, and it left the state unbootable (`record (lonwet-siglyr-hasmer-dovbex/.urb/egz.hope) is corrupt (j)`)

As per @galenwp's request, I've kept the pier and can provide it to anyone interested.

**Comment by joemfb** [comment]
this appears to be at least similar to urbit/arvo#255, maybe even the same bug ...

**Comment by whereswaldon** [comment]
Funny, I commented on that one about how this hadn't caused problems for me... I guess I spoke too soon.

**Comment by joemfb** [comment]
Have you had any subsequent issues with the planet? I'm thinking of `%ap-abut-bad-bone`, as described in urbit/arvo#249.

**Comment by whereswaldon** [comment]
The planet? No. What kind of issues would those be? I hardly use the planet. It just churns up there in AWS.

**Comment by whereswaldon** [comment]
@joemfb The moon side of what you saw seems about right; I saw the same. Sadly, I don't know what my planet did during that time because I wasn't watching it. It could very possibly have done exactly what you saw.

**Comment by whereswaldon** [comment]
Just had a similar problem occur on another moon. I'm still not sure what triggers it. Might be interesting to compare the two piers.

**Comment by cgyarvin** [comment]
Just FYI, the reason I'm not worried about this is that I think it comes
from issues in the transactionality of %gall, which should be fixed in
cc-release.   Also, the pier corruption in the original post is in our
event saving library, which also has a replacement waiting to be
integrated.  Sorry about the crashage!

**Comment by joemfb** [comment]
Isn't it a bit surprising that these crashes happen pretty rarely, except where moons are involved? I've crashed most of my moons in this manner, but haven't run in to these types of bugs otherwise.

**Comment by cgyarvin** [comment]
That is weird.  I think it has to do with the default app configuration of
moons.

**Comment by Fang-** [comment]
Closed alongside urbit/arvo#255
